dvc/repo/fetch.py

```diff
@@ -55,6 +55,8 @@ def stage_filter(stage: "Stage") -> bool:
     def outs_filter(out: "Output") -> bool:
         if push and not out.can_push:
             return False
+        if out.explicit and not any(targets):
```

Comment:
Should `out` be compared to each `target` to see if it matches any of the targets? It looks like this would fetch `out` anytime any `target` is specified, even if the `target` does not match the explicit `out`.

Reply:
I originally went down that path, then it occurred to me that the usual target matching logic would further filter the list that this function returns, so there should be no need to reimplement that logic. I updated the unit test to demonstrate and build confidence.